etest/__init__.py

```diff
@@ -47,9 +47,17 @@ def echo_check_verbose(check):
 @click.option("-j", "--jobs", default=1, help="number of test to run simultaneously")
 @click.option("-q", "--quiet", is_flag=True, default=False, help="suppress all output")
 @click.option("-v", "--verbose", is_flag=True, default=False, help="provide more output")
+@click.option(
+    "-a",
+    "--arch",
+    "--architecture",
+    default="amd64",
+    type=click.Choice(["amd64", "x86", "arm64", "armv5", "armv7", "ppc64"], case_sensitive=False),
+    help="architecture to test against",
+)
```

Comment:
This should have some overlap with etest-build so maybe we should excise it to a separate module.

Reply:
It sounds like a good idea but Im not sure as to how to resolve this.

Reply:
The simple answer would be to make a new module and then create a function to encapsulate this call there and use it as a decorator here.  Just a basic factoring out of this bit.  What do you think?

Reply:
Ill look into it